Sources/KlaviyoSwiftExtension/KlaviyoExtension.swift

```diff
@@ -0,0 +1,133 @@
+//
+//  KlaviyoExtension.swift
+//
+//
+//  Created by Ajay Subramanya on 6/23/23.
+//
+#if !os(macOS)
+import Foundation
+import UserNotifications
+
+public enum KlaviyoExtensionSDK {
+    /// Call this method when you receive a rich push notification in the notification service extension.
+    /// This method should be called from within `didReceive(_:withContentHandler:)` method of `UNNotificationServiceExtension`.
+    /// This method mainly does two things - downloads the media attached in the payload and then attaches it to the push notification.
+    ///
+    /// NOTE that there is no guarantees that the content handler will be called within the iOS stipulated time in the case where the download of the rich media takes long. iOS will automatically
```

Comment:
```NOTE that there is no guarantee that the content handler will be called with in the time stipulated by iOS to download the rich media successfully. In the case where the download does not complete, iOS will automatically present the notification as received from APNS without the attached image```

Just thought that sounded a little clearer. What do you think?

Reply:
Thanks. Reading what I wrote again now makes me want to go hide under a rock!